edward/inferences.py

```diff
@@ -120,34 +138,73 @@ def initialize(self, n_iter=1000, n_data=None, n_print=100,
         init.run()
 
     def update(self):
+        """Run one iteration of optimizer for variational inference.
+    
+        Returns
+        -------
+        loss : double
+            variational `loss function` values after one iteration
+        """
         sess = get_session()
         _, loss = sess.run([self.train, self.loss])
         return loss
 
     def print_progress(self, t, loss):
+        """Print progress to output.
+        
+        Parameters
+        ----------
+        t : int
+            Iteration counter
+        loss : double
+            Loss function value at iteration ``t``
+        """
         if self.n_print is not None:
             if t % self.n_print == 0:
                 print("iter {:d} loss {:.2f}".format(t, loss))
                 print(self.variational)
 
     def finalize(self):
-        """Run steps after all updates."""
+        """Finalize.
+
+        Empty method. (Optional.)
+
+        Any class based on ``VariationalInference`` **may** 
+        implement this method.
+        """
         pass
 
     def build_loss(self):
+        """Build loss function. 
+
+        Empty method. 
+
+        Any class based on ``VariationalInference`` **must** 
+        implement this method.
+
+        Raises
+        ------
+        NotImplementedError
+        """
         raise NotImplementedError()
 
-class MFVI(VariationalInference):
 # TODO this isn't MFVI so much as VI where q is analytic
-    """
-    Mean-field variational inference
-    (Ranganath et al., 2014)
+class MFVI(VariationalInference):
+    """Mean-field variational inference.
+
+    This class implements a variety of "black-box" variational inference
+    techniques (Ranganath et al., 2014) that maximize
+
+    .. math:: 
+
+        ELBO =  E_{q(z; \lambda)} [ \log p(x, z) - \log q(z; \lambda) ]
```

Comment:
How about: "This class [...] minimizes KL(q||p) ... which is equivalent to maximizing ...."?  Part of Edward is teaching people the right way to approach problems, which in this case is focusing on KL(q||p). (you likely agree with me :)